src/main/java/io/confluent/connect/hdfs/avro/AvroHiveUtil.java

```diff
@@ -80,14 +84,15 @@ private Table constructAvroTable(
       String database,
       String tableName,
       Schema schema,
-      Partitioner partitioner
+      Partitioner partitioner,
+      String topic
   )
       throws HiveMetaStoreException {
     Table table = newTable(database, tableName);
     table.setTableType(TableType.EXTERNAL_TABLE);
     table.getParameters().put("EXTERNAL", "TRUE");
-    // tableName is always the topic name
-    String tablePath = hiveDirectoryName(url, config.getTopicsDirFromTopic(tableName), tableName);
+
+    String tablePath = hiveDirectoryName(url, config.getTopicsDirFromTopic(topic), topic);
```

Comment:
I don't think there will be issues with hive, we have tests covering it. The reason to keep them separate is that location is used in other places (e.g. [TopicPartitionWriter](https://github.com/confluentinc/kafka-connect-hdfs/blob/master/src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java#L341)) and besides the extra effort it might be confusing to write to different places depending on whether hive integration is turned on or not.